src/routes/process-user-vocabulary/components/ImageSelectionOverlay.tsx

```diff
@@ -33,39 +42,92 @@ const Container = styled.View`
   justify-content: center;
 `
 
+const Photo = styled.Image`
+  width: ${wp('32%')}px;
+  height: ${wp('32%')}px;
+  margin: ${wp('0.6%')}px;
+`
+
 type ImageSelectionOverlayProps = {
   setVisible: (visible: boolean) => void
-  pushImage: (imageUri: string) => void
+  pushImage: (image: UserVocabularyImage) => void
 }
 
 const ImageSelectionOverlay = ({ setVisible, pushImage }: ImageSelectionOverlayProps): ReactElement => {
   const camera = useRef<RNCamera>(null)
+  const [showGallery, setShowGallery] = useState<boolean>(false)
+  const [photos, setPhotos] = useState<PhotoIdentifier[]>([])
+  const { permissionRequested, permissionGranted } = useGrantPermissions(
+    Platform.OS === 'ios' ? PERMISSIONS.IOS.MEDIA_LIBRARY : PERMISSIONS.ANDROID.READ_MEDIA_IMAGES
```

Comment:
requesting android permissions for media is quite more complicated depending on versions
```js
const androidMediaPermission = Platform.Version >= 33 ? PERMISSIONS.ANDROID.READ_MEDIA_IMAGES : PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE;
```
For Android13 you will also need another permission

On Android 13 the READ_EXTERNAL_STORAGE has been [replace](https://developer.android.com/about/versions/13/behavior-changes-13#granular-media-permissions) by READ_MEDIA_IMAGES and READ_MEDIA_VIDEO.
```js
<manifest>
...
<uses-permission android:name="android.permission.READ_MEDIA_IMAGES"/>
<uses-permission android:name="android.permission.READ_MEDIA_VIDEO"/>
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
...
<application>
```
Also check for
```js
<application
      android:requestLegacyExternalStorage="true"
      ```